internal/provider/namespace_resource.go

```diff
@@ -278,21 +284,35 @@ func (r *namespaceResource) Create(ctx context.Context, req resource.CreateReque
 			return
 		}
 	}
-	mtls := &namespacev1.MtlsAuthSpec{}
-	if plan.AcceptedClientCA.ValueString() != "" {
-		mtls.Enabled = true
-		mtls.AcceptedClientCa = plan.AcceptedClientCA.ValueString()
-		mtls.CertificateFilters = certFilters
+
+	var spec = &namespacev1.NamespaceSpec{
+		Name:          plan.Name.ValueString(),
+		Regions:       regions,
+		RetentionDays: int32(plan.RetentionDays.ValueInt64()),
+		CodecServer:   codecServer,
 	}
+
+	if plan.ApiKeyAuth.ValueBool() {
+		spec.ApiKeyAuth = &namespacev1.ApiKeyAuthSpec{Enabled: true}
```

Comment:
I wonder if we should error out when both ApiKeyAuth and AcceptedClientCA are set, rather then ignoring the CA cert silently, might avoid some confusion.

Reply:
done